woocommerce/woocommerce
# [CYS- Core] Improve default palette

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



This PR is necessary because we want to have the same style of TT4 for the buttons.

By default, all the palettes have customization for the `core/button` ([source code](https://github.com/woocommerce/woocommerce/blob/46dcb4fd65e4b9824a89375c2eea90910e198bfb/plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/global-styles/color-palette-variations/constants.ts/#L1423-L1437)). For the default palette, instead, we want to rely only on the the `elements` setting ([source code](https://github.com/woocommerce/woocommerce/blob/46dcb4fd65e4b9824a89375c2eea90910e198bfb/plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/global-styles/color-palette-variations/constants.ts/#L43-L56)): otherwise it looks impossible reproduce the `TT4` hover effect.

For this reason, an empty object is assigned `core/button`. 

During the test, I noticed that the `core/button` value was populated again during the switch from another palette. This was caused from `mergeBaseAndUserConfigs`: this function is just a wrapper around deepmerge library: https://github.com/WordPress/gutenberg/blob/237865fad0864c209a7c3e771e23fe66f4fbca25/packages/edit-site/src/components/global-styles/global-styles-provider.js/#L24-L31
Deepmerge library merges two objects, x and y, deeply, returning a new merged object with the elements from both x and y.
In the case of the `variation.title === 'New - Neutral'`, the core/button is an empty object, because we don't want that the classes for the core/button are created.
Deepmerge merges the `userStyles.blocks[ 'core/button' ]` with the `variation.styles.blocks[ 'core/button']` and the result is an object with values that don't match the variation. For this reason, it is necessary to remove the `userStyles.blocks[ 'core/button' ]`.






Closes #44685.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new WooCommerce installation with this version.
2. Ensure the WooCommerce Beta Tester plugin is installed and activated (available on this monorepo).
3. Head over to `/wp-admin/tools.php?page=woocommerce-admin-test-helper` and enable `customize-store` feature flag.
4. Install this build of Gutenberg: [gutenberg.zip](https://github.com/woocommerce/woocommerce/files/14236920/gutenberg.zip)
5. Visit the `wp-admin/admin.php?page=wc-admin&path=/customize-store`.
6. Follow the process.
7. Click on `Change your colors`.
8. Ensure that the first palette is selected.
9. Check the `Shop the sale` color button and ensure that is dark.
10. Switch to another palette.
11. Switch again to the first palette.
12. Ensure that the `Shop the sale` color button is dark.
13. Save.
14. Click done.
15. Visit the frontend.
16. Hover the `shop the sale` button and ensure that the button looks like this:
![image](https://github.com/woocommerce/woocommerce/assets/4463174/14d2efb9-5c37-421c-82df-49f3b2f88c5e)
17. Visit the shop page.
18. Hover a button.
19. Ensure that the button color looks like the previous one.


Kudos to @danielwrobert for pair session!




### Changelog entry




<details>

#### Significance




#### Type




#### Message 
[CYS- Core] Improve the default palette.

#### Comment 

</details>
